src/app/data/AdService.jsx

```diff
@@ -0,0 +1,83 @@
+"use server";
+
+import { notFound } from "next/navigation";
+import { getDefaultHeaders } from "@/app/_common/utils/fetch";
+import mapAdData from "./AdData";
+
+// Expose only necessary data to client
+const sourceIncludes = [
+    "businessName",
+    "contactList.email",
+    "contactList.name",
+    "contactList.phone",
+    "contactList.title",
+    "employer.name",
+    "employer.locationList.postalCode",
+    "employer.locationList.city",
+    "employer.locationList.address",
+    "employer.locationList.municipal",
+    "employer.locationList.country",
+    "employer.location.city" /* todo finnes employer dataene lengre?*/,
+    "employer.location.county" /* todo finnes employer dataene lengre?*/,
+    "employer.location.country" /* todo finnes employer dataene lengre?*/,
+    "expires",
+    "id",
+    "locationList.postalCode",
+    "locationList.city",
+    "locationList.address",
+    "locationList.municipal",
+    "locationList.county",
+    "locationList.country",
+    "medium",
+    "properties.adtext",
+    "properties.address",
+    "properties.applicationdue",
+    "properties.applicationemail",
+    "properties.applicationurl",
+    "properties.employer",
+    "properties.employerdescription",
+    "properties.employerhomepage",
+    "properties.engagementtype",
+    "properties.extent",
+    "properties.facebookpage",
+    "properties.hasInterestform",
+    "properties.jobarrangement",
+    "properties.jobpercentage",
+    "properties.jobtitle",
+    "properties.linkedinpage",
+    "properties.location",
+    "properties.positioncount",
+    "properties.remote",
+    "properties.sector",
+    "properties.sourceurl",
+    "properties.starttime",
+    "properties.twitteraddress",
+    "properties.workday",
+    "properties.workhours",
+    "properties.workLanguage",
+    "published",
+    "reference",
+    "source",
+    "status",
+    "title",
+    "updated",
+].join(",");
+
+export async function getAdData(id) {
+    const res = await fetch(
+        `${process.env.PAMSEARCHAPI_URL}/stillingsok/ad/ad/${id}?_source_includes=${sourceIncludes}`,
+        {
+            headers: getDefaultHeaders(),
+            next: { revalidate: 60 },
+        },
+    );
+
+    if (res.status === 404) {
+        notFound();
+    }
```

Comment:
En ren data-layer borde inte säga åt ramverket (Next.js) vad det ska göra när vi inte klarar av att hämta data. Vad med att returnera `undefined` eller `null` och låta returnera en `404 Not Found` i page i stället?

Reply:
For å kunne skille på error og not found - hva med å heller kaste en error?
For eksempel noe sånn:

`throw new Error("Ad not found", { reason: "not_found" });`

eller hvis feil:
`throw new Error("Failed to fetch ad", { reason: "error" });`

Reply:
Antingen det eller returnera ett objekt som kan innehålla både ok och error/status. I just det här tillfället fungerar det väl att kasta en error, men annars tycker man borde undvika try-catch-errors för att styra logik. Det kan fort bli lite oöversiktligt.

Reply:
Antingen det eller returnera ett objekt som kan innehålla både ok och error/status. I just det här tillfället fungerar det väl att kasta en error, men annars tycker man borde undvika try-catch-errors för att styra logik. Det kan fort bli lite oöversiktligt.

Reply:
det blir noe sånn her da
```
{
  success: true,
  data: <adData>,
}
```
```
{
  success: false,
  error: "not_found",
}
```